jlm/llvm/opt/DeadNodeElimination.cpp

```diff
@@ -259,22 +259,25 @@ DeadNodeElimination::MarkOutput(const jlm::rvsdg::output & output)
     }
   }
 
-  if (auto phiOutput = dynamic_cast<const phi::rvoutput *>(&output))
+  if (auto phi = rvsdg::TryGetOwnerNode<rvsdg::PhiNode>(output))
   {
-    MarkOutput(*phiOutput->result()->origin());
+    MarkOutput(*phi->MapOutputFixVar(output).result->origin());
     return;
   }
 
-  if (auto phiRecursionArgument = dynamic_cast<const phi::rvargument *>(&output))
+  if (auto phi = rvsdg::TryGetRegionParentNode<rvsdg::PhiNode>(output))
   {
-    MarkOutput(*phiRecursionArgument->result()->origin());
-    return;
-  }
-
-  if (auto phiInputArgument = dynamic_cast<const phi::cvargument *>(&output))
-  {
-    MarkOutput(*phiInputArgument->input()->origin());
-    return;
+    auto var = phi->MapArgument(output);
+    if (auto fix = std::get_if<rvsdg::PhiNode::FixVar>(&var))
+    {
+      MarkOutput(*fix->result->origin());
+      return;
+    }
+    else if (auto ctx = std::get_if<rvsdg::PhiNode::ContextVar>(&var))
+    {
+      MarkOutput(*ctx->input->origin());
+      return;
+    }
   }
```

Comment:
an else with unreachable to exhaust all possibilities?

Reply:
added